realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -641,6 +641,13 @@ public RealmFieldType getFieldType(String fieldName) {
         return proxyState.getRow$realm().getColumnType(columnIndex);
     }
 
+    /**
+     * Currently, the hash value returned from this could suddenly change as the object's index might
+     * become different along with the changes of the Realm. Thus, it is  advisable not to solely rely
+     * on the hash value to identify an instance of {@link DynamicRealmObject}.
```

Comment:
Instead of completely replacing the default javadoc for hashcode I would reuse the standard one and then add our caveats. Something like:

```
     * Returns an integer hash code for this object. By contract, any two
     * objects for which {@link #equals} returns {@code true} must return
     * the same hash code value. 
     *
     * <p>Note that because RealmObjects are "live" objects, they might be updated by writes from other threads. 
     * This means that the hash code is not stable and RealmObjects should not be used as keys in HashMap saved in a HashSet.
     *
     * @return this object's hash code.
     * @see #equals